src/components/ExplorerPage.tsx

```diff
@@ -0,0 +1,376 @@
+import { abbreviateHash } from "../util/addresses";
+import { constants } from "../util/helpers";
+import { sendErrorNotification } from "../util/sendErrorNotification";
+import { PageContent } from "./formatting";
+import { AddressTagInput } from "./AddressTagInput";
+import { UpdateAddressTagsModal } from "./UpdateAddressTagsModal";
+
+const { ADDRESSES_PER_PAGE } = constants;
+const { Option } = Select;
+const CHAINS = {
+  BITCOIN_LEGACY: "Bitcoin (Legacy)",
+  BITCOIN_SEGWIT: "Bitcoin (Segwit)",
+  BITCOIN_WRAPPED_SEGWIT: "Bitcoin (Wrapped Segwit)",
+  ETHEREUM: "Ethereum",
+  SOLANA: "Solana",
+  CUSTOM: "Custom",
+};
+
+const ExplorerPage = () => {
+  const { addressTags } = useAddressTags();
+  const { session } = useContext(AppContext);
+  const [addresses, setAddresses] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isAddAddressesModalVisible, setIsAddAddressesModalVisible] =
+    useState(false);
+  // m / purpose' / coin_type' / account' / change / index
+  const [purpose, setPurpose] = useState(44);
+  const [isPurposeHardened, setIsPurposeHardened] = useState(true);
+  const [coinType, setCoinType] = useState(60);
+  const [isCoinTypeHardened, setIsCoinTypeHardened] = useState(true);
+  const [account, setAccount] = useState(0);
+  const [isAccountHardened, setIsAccountHardened] = useState(true);
+  const [change, setChange] = useState(0);
+  const [isChangeHardened, setIsChangeHardened] = useState(false);
+  const [index, setIndex] = useState(0);
+  const [isIndexHardened, setIsIndexHardened] = useState(false);
+  const [selectedChain, setSelectedChain] = useState(CHAINS.ETHEREUM);
+
+  const getInitialAddressTags = () => {
+    return addresses.map((addr) => {
+      if (addr.record && addr.record.id) {
+        return { key: addr.record.key, val: addr.record.val };
+      }
+      return { key: addr.address, val: "" };
+    });
+  };
+
+  const getPath = () => {
+    const HARDENED_OFFSET = 0x80000000;
+    const purposePath = isPurposeHardened ? HARDENED_OFFSET + purpose : purpose;
+    const coinTypePath = isCoinTypeHardened
+      ? HARDENED_OFFSET + coinType
+      : coinType;
+    const accountPath = isAccountHardened ? HARDENED_OFFSET + account : account;
+    const changePath = isChangeHardened ? HARDENED_OFFSET + change : change;
+    const indexPath = isIndexHardened ? HARDENED_OFFSET + index : index;
+    const path = [
+      purposePath,
+      coinTypePath,
+      accountPath,
+      changePath,
+      indexPath,
+    ].filter((x) => x !== null);
+    return path;
+  };
+
+  const getAddrs = () => {
+    setIsLoading(true);
+    const flag = selectedChain === CHAINS.SOLANA ? 4 : undefined;
+    session.client
+      .getAddresses({
+        startPath: getPath(),
+        n: ADDRESSES_PER_PAGE,
+        flag,
+      })
+      .then((addrs) => {
+        setAddresses(
+          addrs.map((addr) => ({
+            address:
+              selectedChain === CHAINS.SOLANA
```

Comment:
Also just want to make it clear... you can still derive an e.g. ETH address from a number of other `coin_types` than the main Ethereum one. I think this param would be better named `selectedDisplayType`, which can be e.g. `ETH_address`, `SOL_address`, `BLS12_381_pubkey`, etc. Using `selectedChain` is a bit mismatched with what is actually being derived, which is an address or pubkey based on the selected *format*.

Functionally what you have is fine because you have to land on a chain type based on your dropdown, but it's a bit confusing to read here.

Reply:
[19aaad1](https://github.com/GridPlus/lattice-manager/pull/287/commits/19aaad1cb1fd0590257c0fca5f49d8ae201acd48)

Reply:
That makes sense. I wasn't thinking about it from that perspective. 

I added a switch for flag and the string transforms, so we can expand it out easily.